geopandas/io/tests/test_arrow.py

```diff
@@ -776,11 +795,19 @@ def test_write_read_to_pandas_kwargs(tmpdir, format):
         gdf.to_parquet(filename)
         read_func = read_parquet
 
+    # Note: due to bugs in pyarrow, we can't read complex types without using
+    # the types mapper
+    match = re.escape("data type 'struct<foo: string>[pyarrow]' not understood")
+    with pytest.raises(TypeError, match=match):
+        read_func(filename)
```

Comment:
Do you mind moving this to a seperate test and marking as xfail? I'd rather have it so that a fix in pyarrow doesn't break the CI here. Also, do you happen to know if this is reported upstream so we can include a note on the issue number?